src/contrib/mlir/backend/pass/affine_lowerer.cpp

```diff
@@ -537,9 +603,548 @@ namespace
+                createStore(llvmI8Ty,
+                            builder.getI8IntegerAttr(static_cast<int8_t>(
+                                attrs.poolAttrs2d.includePaddingInAvgComputation)),
+                            geps[0],
+                            builder);
+                for (auto k = 0; k < 2; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[1],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs2d.windowShape[k]),
+                                gepStructOp,
+                                builder);
+                }
+                for (auto k = 0; k < 2; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[2],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs2d.windowStrides[k]),
+                                gepStructOp,
+                                builder);
+                }
+                for (auto k = 0; k < 2; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[3],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs2d.padBelow[k]),
+                                gepStructOp,
+                                builder);
+                }
+                for (auto k = 0; k < 2; k++)
+                {
+                    auto gepStructOp =
+                        builder.create<LLVM::GEPOp>(builder.getUnknownLoc(),
+                                                    llvmI64PtrTy,
+                                                    geps[4],
+                                                    ArrayRef<Value>({constants[0], constants[k]}));
+                    createStore(llvmI64Ty,
+                                builder.getI64IntegerAttr(attrs.poolAttrs2d.padAbove[k]),
+                                gepStructOp,
+                                builder);
+                }
+                break;
+            }
+            case AttrsType::POOL3D:
```

Comment:
If this is just introducing support for ConvBias, why do we need all these code for other ops? Is it possible to leave that for a future patch?

Reply:
We have to include those because we're changing the way we store and pass attributes to callbacks.